vitest.shared.mts

```diff
@@ -36,24 +36,9 @@ export const alias = [
 ];
 
 export default defineConfig({
-  plugins: [
-    react({
-      babel: {
-        plugins: [
-          [
-            '@mui/internal-babel-plugin-display-name',
-            {
-              allowedCallees: {
-                '@mui/x-internals/forwardRef': ['forwardRef'],
-              },
-            },
-          ],
-        ],
-        babelrc: false,
-        configFile: false,
-      },
-    }),
-  ],
+  // If enabling babel plugins, ensure the tests in CI are stable
+  // https://github.com/mui/mui-x/pull/18341
+  plugins: [react()],
```

Comment:
If this was the culprit for almost 100% flaky tests, I'm all in with the compromise. 👍 
Great work finding the root cause! 💙 💯